src/lib.rs

```diff
@@ -1368,7 +1368,8 @@ impl<TRepoTypes: RepoTypes> Future for IpfsFuture<TRepoTypes> {
         loop {
             loop {
                 let inner = {
-                    let next = self.swarm.next_event();
+                    use futures::StreamExt;
+                    let next = self.swarm.next();
```

Comment:
Given that this is an infinite stream, you might as well use https://docs.rs/futures/0.3.16/futures/stream/trait.StreamExt.html#method.select_next_some .

Reply:
Fair point, especially that indeed this is an [infinite stream](https://docs.rs/libp2p/0.39.1/libp2p/swarm/struct.ExpandedSwarm.html#impl-Stream).

Reply:
https://github.com/rs-ipfs/rust-ipfs/pull/472/commits/87aaefd616d21316146ddd8675a925104790b7ad